### PR TITLE
Implement easing

### PR DESCRIPTION
**Pull request by foldfelis**

### CODE REVIEW

**Comment by foldfelis** [comment]
please rebase `dev` onto `main` after merge this

File: src/easing.jl [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+    return EaseBounceOut(2t-d, 0, c, d)/2 + c/2 + b
+end
+
+# ############################
+# # Elastic Easing functions #
+# ############################
+
+function EaseElasticIn(t::Real, b::Real, c::Real, d::Real)
+    (t ≈ 0) && (return b)
+    ((t/=d) ≈ 1) && (return b + c)
```
Comment (chengchingwen) [inline src/easing.jl:201]:
```suggestion
    (t ≈ d) && (return b + c)
    t = 1
```

File: src/easing.jl [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+    p = 0.3d
+    a = c
+    s = p/4
+
+    return a * 2^(-10t) * sin((t * d - s) * 2π/p) + c + b
+end
+
+function EaseElasticInOut(t::Real, b::Real, c::Real, d::Real)
+    (t ≈ 0) && (return b)
+    ((t /= d/2) ≈ 2) && (return b + c)
```
Comment (chengchingwen) [inline src/easing.jl:224]:
```suggestion
    (t ≈ d) && (return b + c)
    t = 2
```

Comment (chengchingwen) [inline src/easing.jl:224]:
```suggestion
    (t ≈ d) && (return b + c)
    t = 2
```

File: src/easing.jl [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+    a = c
+    s = p/4
+    postFix = a * 2^(10(t -= 1))
+
+    return -(postFix * sin((t * d - s) * 2π/p )) + b
+end
+
+function EaseElasticOut(t::Real, b::Real, c::Real, d::Real)
+    (t ≈ 0) && (return b)
+    ((t /= d) ≈ 1) && (return b + c)
```
Comment (chengchingwen) [inline src/easing.jl:213]:
```suggestion
    (t  ≈ d) && (return b + c)
    t = 1
```

**Review by chengchingwen — CHANGES_REQUESTED** [review]

**Review by chengchingwen — APPROVED** [review]

**Pull request merged**